packages/frontend/src/pages/Executions/components/SearchWithFilterInput.tsx

```diff
@@ -0,0 +1,99 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { BiSearch, BiSolidXCircle } from 'react-icons/bi'
+import {
+  Divider,
+  Icon,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputRightElement,
+} from '@chakra-ui/react'
+import { debounce } from 'lodash'
+
+import ExecutionStatusMenu from '@/components/ExecutionStatusMenu'
+
+interface SearchWithFilterInputProps {
+  status: string
+  searchValue: string
+  onChange: (val: string) => void
+  onStatusChange: (newStatus: string) => void
+}
+
+export default function SearchWithFilterInput({
+  searchValue,
+  status,
+  onChange,
+  onStatusChange,
+}: SearchWithFilterInputProps) {
+  const filterRef = useRef<HTMLDivElement>(null)
+  const [inputPadding, setInputPadding] = useState<number>(0)
+  const [tempSearchValue, setTempSearchValue] = useState(searchValue)
+
+  const handleSearchInputChangeDebounced = useMemo(
+    () => debounce(onChange, 500),
+    [onChange],
+  )
+
+  const onSearchInputChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setTempSearchValue(event.target.value)
+      handleSearchInputChangeDebounced(event.target.value)
+    },
+    [handleSearchInputChangeDebounced],
+  )
+
+  const clearSearch = useCallback(() => {
+    setTempSearchValue('')
+    onChange('')
+    handleSearchInputChangeDebounced.cancel()
+  }, [onChange, handleSearchInputChangeDebounced])
+
+  // update padding of input element when filter element width changes.
+  useEffect(() => {
+    if (!filterRef.current) {
+      return
+    }
+    setInputPadding(filterRef.current.offsetWidth + 8)
+  }, [status])
```

Comment:
`[filterRef.current?.offsetWidth, status]`, seems like need to add the offsetWidth check or else when first typing, it doesn't update the padding properly hmm